src/dmd/dinterpret.d

```diff
@@ -5019,6 +5019,33 @@ public:
                 result = interpret(ae, istate);
                 return;
             }
+            else if (fd.ident == Id._d_arraycatnTX)
+            {
+                assert(e.arguments.dim == 1);
+
+                Expression ea = (*e.arguments)[0];
+
+                while (true)
```

Comment:
You have used a simpler `while(ea.isCastExp()) ea = ea.isCastExp().e1;` elsewhere.